src/LtiServiceConnector.php

```diff
@@ -172,6 +194,47 @@ public function getAll(
         return $results;
     }
 
+    public function logRequest(
+        int $requestType,
+        IServiceRequest $request,
+        array $responseHeaders,
+        ?array $responseBody
+    ): void {
+        $contextArray = [
+            'request_method' => $request->getMethod(),
+            'request_url' => $request->getUrl(),
+            'response_headers' => $responseHeaders,
+            'response_body' => json_encode($responseBody),
+        ];
+        $userId = '';
+
+        $payload = $request->getPayload();
+        if (isset($payload['body'])) {
+            $requestBody = $payload['body'];
+
+            $contextArray['request_body'] = $requestBody;
+
+            if (isset(json_decode($requestBody)->userId)) {
+                $userId = json_decode($requestBody)->userId;
+            }
+        }
+
+        $this->errorLog($requestType, $userId, $contextArray);
```

Comment:
Mmm, I intended this function (`logRequest()`) to only be called by the `LtiServiceConnector` and check for `$this->debuggingMode` before it's called, but because this function was `public` that intent wasn't clear - I've made it `private` to clarify that, which should make it so that this only gets logged if `$this->debuggingMode` is `true`.